src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -1222,486 +1624,613 @@ s32 func_80A13B1C(EnGo* this, PlayState* play) {
-            func_80A12C48(this, play, 7);
-            this->unk_3C0++;
+            EnGo_ChangeAnim(this, play, ENGO_ANIM_DROPKEG);
+            this->cutsceneState++;
 
         case 2:
             if (Animation_OnFrame(&this->skelAnime, 16.0f)) {
                 Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_GORON_HAND_HIT);
             }
 
             if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
-                func_80A12C48(this, play, 1);
-                Lib_Vec3f_TranslateAndRotateY(&this->actor.world.pos, this->actor.shape.rot.y, &D_80A166A4, &sp48);
+                EnGo_ChangeAnim(this, play, ENGO_ANIM_LYINGDOWNIDLE_IMM);
+                Lib_Vec3f_TranslateAndRotateY(&this->actor.world.pos, this->actor.shape.rot.y, &sPowderKegSpawnOffset,
+                                              &powderKegSpawnPos);
                 gSaveContext.powderKegTimer = 2400;
-                Actor_Spawn(&play->actorCtx, play, ACTOR_EN_BOM, sp48.x, sp48.y, sp48.z, 1, 0, 0, 0);
-                func_80A134B0(this, play, 1);
-                this->unk_3C2 = 0;
-                this->unk_3C0++;
+                Actor_Spawn(&play->actorCtx, play, ACTOR_EN_BOM, powderKegSpawnPos.x, powderKegSpawnPos.y,
+                            powderKegSpawnPos.z, 1, 0, 0, 0);
+                EnGo_PowderKegGoron_UpdateColliderRadius(this, play, true);
+                this->cutsceneDelayTimer = 0;
+                this->cutsceneState++;
             }
             break;
 
         case 3:
-            if (this->unk_3C2 >= 60) {
-                ActorCutscene_Stop(this->unk_3B8);
-                this->unk_3C2 = 0;
-                this->unk_3C0 = 0;
+            if (this->cutsceneDelayTimer >= 60) {
+                ActorCutscene_Stop(this->indexCutscene);
+                this->cutsceneDelayTimer = 0;
+                this->cutsceneState = 0;
                 ret = true;
             } else {
-                this->unk_3C2++;
+                this->cutsceneDelayTimer++;
             }
             break;
     }
 
     return ret;
 }
 
-void func_80A14018(EnGo* this, PlayState* play) {
-    static Vec3f D_80A166B0 = { 0.0f, 0.0f, 40.0f };
-    static s32 D_80A166BC[] = { 11, 10, 12, 13, 14, 17 };
-    Vec3f sp2C;
-    s32 phi_v0 = ENGO_GET_70(&this->actor) % 6;
+/**
+ * Update the stretching racer's animation and action flags
+ *
+ * EnGoGoronRaceSubtypes A-D exchange animations before and after the race, while the Hamstring Stretchers
+ * have the same animation every time.
+ *
+ * @see EnGo_Idle
+ */
+void EnGo_UpdateToStretchingAnimation(EnGo* this, PlayState* play) {
+    static Vec3f sStretchingGoronOffset = { 0.0f, 0.0f, 40.0f };
+
+    static s32 sSubtypeToAnimIndex[] = { ENGO_ANIM_SQUAT_SIDE_TO_SIDE, ENGO_ANIM_DOUBLE_ARM_SIDEBEND,
+                                         ENGO_ANIM_SHAKE_LIMBS,        ENGO_ANIM_SINGLE_ARM_SIDEBEND,
+                                         ENGO_ANIM_SITTING_STRETCH,    ENGO_ANIM_HELP_SITTING_STRETCH };
```

Comment:
```suggestion
    static s32 sSubtypeToAnimIndex[] = { ENGO_ANIM_SQUAT_SIDE_TO_SIDE, ENGO_ANIM_DOUBLE_ARM_SIDEBEND,
                                         ENGO_ANIM_SHAKE_LIMBS,        ENGO_ANIM_SINGLE_ARM_SIDEBEND,
                                         ENGO_ANIM_SITTING_STRETCH,    ENGO_ANIM_HELP_SITTING_STRETCH, };
```
Should format better with a trailing comma.